AlfrescoApp/Model/URLHandlers/AlfrescoURLHandler.m

```diff
@@ -0,0 +1,281 @@
+    if(numberOfAccountsSetup > 0)
+    {
+        if(numberOfAccountsSetup > 1)
+        {
+            NSError *keychainError = nil;
+            NSArray *savedAccounts = [KeychainUtils savedAccountsForListIdentifier:kAccountsListIdentifier error:&keychainError];
+            
+            if (keychainError)
+            {
+                AlfrescoLogError(@"Error accessing shared keychain. Error: %@", keychainError.localizedDescription);
+            }
+            
+            // Create wrapper accounts
+            NSArray *wrapperAccounts = [self createAlfrescoKitUserAccountsFromAppAccounts:savedAccounts];
+            // Display the accounts controller
+            AKUserAccountListViewController *userAccountViewController = [[AKUserAccountListViewController alloc] initWithAccountList:wrapperAccounts delegate:self];
+            UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:userAccountViewController];
+            SwitchViewController *switchController = [self getPresentingViewController];
+            if(switchController)
+            {
+                [UniversalDevice displayModalViewController:navigationController onController:switchController withCompletionBlock:nil];
+            }
+        }
+        else
+        {
+            handled = [self handleURL:url session:session];
+        }
+    }
+
+    
+    return handled;
+}
+
+#pragma mark - Private methods
+- (AlfrescoURLType) parseURLForAction: (NSString *)URLString
+{
+    //Removing the scheme from the url string
+    NSString *urlWithoutScheme = [URLString stringByReplacingOccurrencesOfString:kHandlerPrefix withString:@""];
+    
+    if([urlWithoutScheme hasPrefix:kDocumentPath])
+    {
+        return AlfrescoURLTypeDocument;
+    }
+    else if ([urlWithoutScheme hasPrefix:kFolderPath])
+    {
+        return AlfrescoURLTypeFolder;
+    }
+    else if ([urlWithoutScheme hasPrefix:kSitePath])
+    {
+        return AlfrescoURLTypeSite;
+    }
+    else if ([urlWithoutScheme hasPrefix:kUserPath])
+    {
+        return AlfrescoURLTypeUser;
+    }
+    
+    return AlfrescoURLTypeNone;
+}
+
+- (void)presentViewControllerFromURL:(UIViewController *)controller
+{
+    SwitchViewController *switchController = [self getPresentingViewController];
+    if(switchController)
+    {
+        NavigationViewController *navigationController = [[NavigationViewController alloc] initWithRootViewController:controller];
+        [switchController displayURLViewController:navigationController];
+    }
+    
+    if([[UniversalDevice rootMasterViewController] isKindOfClass:[MainMenuViewController class]])
+    {
+        MainMenuViewController *mainMenu = (MainMenuViewController *)[UniversalDevice rootMasterViewController];
+        [mainMenu cleanSelection];
+    }
+}
+
+- (SwitchViewController *)getPresentingViewController
+{
+    if([[UniversalDevice rootDetailViewController] isKindOfClass:[DetailSplitViewController class]])
```

Comment:
As it stands today yes I think that IS_IPAD could be used but I think this condition would have to be changed once support for iPhone6+ is added.